File: wear/src/main/java/com/androidessence/cashcaretaker/Account.java
package com.androidessence.cashcaretaker;

import android.content.ContentValues;

import org.json.JSONException;
import org.json.JSONObject;

/**
 * Created by adammcneilly on 12/27/15.
 */
public class Account {
    private long identifier;
    private String name;
    private double balance;

    public Account(long identifier, String name, double balance) {
        this.identifier = identifier;
        this.name = name;
        this.balance = balance;
    }

    public Account(JSONObject jsonObject) throws JSONException{
        this.identifier = jsonObject.getLong(CCContract.AccountEntry._ID);
        this.name = jsonObject.getString(CCContract.AccountEntry.COLUMN_NAME);
        this.balance = jsonObject.getDouble(CCContract.AccountEntry.COLUMN_BALANCE);
    }

    public long getIdentifier() {
        return identifier;
    }

    public void setIdentifier(long identifier) {
        this.identifier = identifier;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public double getBalance() {
        return balance;
    }

    public void setBalance(double balance) {
        this.balance = balance;
    }

    public ContentValues getContentValues() {
        ContentValues values = new ContentValues();

        values.put(CCContract.AccountEntry._ID, getIdentifier());
        values.put(CCContract.AccountEntry.COLUMN_NAME, getName());
        values.put(CCContract.AccountEntry.COLUMN_BALANCE, getBalance());

        return values;
    }
}

File: app/src/main/java/com/androidessence/cashcaretaker/refresh/Account.kt
package com.androidessence.cashcaretaker.refresh

import android.content.ContentValues
import android.database.Cursor
import android.os.Parcel
import com.androidessence.cashcaretaker.creator
import com.androidessence.cashcaretaker.data.CCContract

/**
 * Represents a bank account.
 *
 * Created by adam.mcneilly on 1/25/17.
 */
open class Account: BaseModel {
    var name = ""
    var balance: Double = 0.toDouble()

    constructor(): super()

    constructor(cursor: Cursor) {
        name = cursor.getString(cursor.getColumnIndex(CCContract.AccountEntry.COLUMN_NAME))
        balance = cursor.getDouble(cursor.getColumnIndex(CCContract.AccountEntry.COLUMN_BALANCE))
    }

    constructor(source: Parcel): super(source) {
        name = source.readString()
        balance = source.readDouble()
    }

    override fun writeToParcel(dest: Parcel?, flags: Int) {
        super.writeToParcel(dest, flags)

        dest?.writeString(name)
        dest?.writeDouble(balance)
    }

    override fun getContentValues(): ContentValues {
        val values = super.getContentValues()

        values.put(CCContract.AccountEntry.COLUMN_NAME, name)
        values.put(CCContract.AccountEntry.COLUMN_BALANCE, balance)

        return values
    }

    companion object {
        @JvmField val CREATOR = creator(::Account)
    }
}